closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1070,23 +887,615 @@ boolean isRestrictedProviderAllowed(String providerName) {
+            rsProperties.descPolicy = getProperty("descPolicy");
+            rsProperties.descSunsetDate = getProperty("descSunsetDate");
+
+            // Security properties.
+            rsProperties.jdkTlsDisabledNamedCurves = getProperty("jdkTlsDisabledNamedCurves");
+            rsProperties.jdkTlsDisabledAlgorithms = getProperty("jdkTlsDisabledAlgorithms");
+            rsProperties.jdkTlsEphemeralDHKeySize = getProperty("jdkTlsEphemeralDHKeySize");
+            rsProperties.jdkTlsLegacyAlgorithms = getProperty("jdkTlsLegacyAlgorithms");
+            rsProperties.jdkCertpathDisabledAlgorithms = getProperty("jdkCertpathDisabledAlgorithms");
+            rsProperties.jdkSecurityLegacyAlgorithms = getProperty("jdkSecurityLegacyAlgorithms");
+            rsProperties.keyStoreType = getProperty("keyStoreType");
+            rsProperties.keyStore = getProperty("keyStore");
+
+            // For SecureRandom.
+            rsProperties.jdkSecureRandomProvider = getProperty("jdkSecureRandomProvider");
+            rsProperties.jdkSecureRandomAlgorithm = getProperty("jdkSecureRandomAlgorithm");
+
+            rsProperties.jdkFipsMode = getProperty("jdkFipsMode");
+
+            rsProperties.providers.addAll(this.providers);
+            rsProperties.providersSimpleName.addAll(this.providersSimpleName);
+            rsProperties.providerConstraints.putAll(this.providerConstraints.entrySet()
+                                                                            .stream()
+                                                                            .collect(Collectors.toMap(
+                                                                                e -> e.getKey(),
+                                                                                e -> e.getValue().toArray(new Constraint[0])
+                                                                            )));
+
+            if (debug != null) {
+                // Print information of utilized security profile.
+                rsProperties.listUsedProfile();
+            }
+
+            return rsProperties;
+        }
+
+        /**
+         * Initialize restricted security properties.
+         */
+        private void init(String profileID) {
+            if (debug != null) {
+                debug.println("Initializing restricted security properties for '" + profileID + "'.");
+            }
+
+            String potentialExtendsProfileID = parseProperty(securityProps.getProperty(profileID + ".extends"));
+            if (potentialExtendsProfileID != null) { // If profile extends another profile.
+                if (debug != null) {
+                    debug.println("\t'" + profileID + "' extends '" + potentialExtendsProfileID + "'.");
+                }
+
+                // Check if extended profile exists.
+                String extendsProfileID = null;
+                if (potentialExtendsProfileID.indexOf(".") != potentialExtendsProfileID.lastIndexOf(".")) {
```

Comment:
Prefer `indexof(char)` over `indexOf(String)` when the argument is one character.